codeChunks.Rmd

```diff
@@ -0,0 +1,156 @@
+```{r eval=FALSE}
+# grab data paths from the folder that data is stored in 
+# "path" specifies name of folder that data paths are stored 
+# full.names = T produces full paths to fukes instead of just file name
+
+rawPaths <- dir(path = "path/to/folder", full.names = T) 
+```
+
+Read in data using a for loop. Remember to initialize all variables that you will be using outside of the for loop. 
+```{r eval=FALSE}
+dataList <- vector("list", length(rawPaths)) # makes an empty list with same length as file paths vector
+i=0 
+for(i in 1:length(rawPaths)){
+  dataList[[i]] <- read.table(rawPaths[i], 
+                               na.strings = c("", "NA"),
+                               header=T)  
+}
+```
+Note: list() creates an empty list of length 0. However, vector("list", length(rawPaths)) allocates a designated number of slots within the list instead of the list being constantly updated every time the for loop interates. With a small number of iterations, the time it takes for the code to run is not noticeable. However, for a large number of iterations, not allocating space will cause the code to run very slowly. 
+
+
+#Removing Extraneous Rows and Columns
+
+##Rows
+
+Iterate through all the rows in a data frame. 
+allRows is a vector containing "TRUE" and "FALSE". Each element corresponds to a row in dataFrame. 
+is.na(dataFrame[i,]) outputs "TRUE" if the row contains at least one blank cell, and "FALSE" otherwise. 
+all(is.na(dataFrame[i,])) outputs "TRUE" if all cells in that row are blank, and "FALSE" otherwise. 
+```{r eval=F}
+i=0
+allRows <- c() # initialize vector 
+for(i in 1:nrow(dataFrame)){
+  allRows[i] <- all(is.na(dataFrame[i,])) # store each output into allRows
+}
+
+blankRows <- which(allRows) # outputs indices of rows that contain "TRUE" (rows with all NA's) 
+dataFrame <- dataFrame[-blankRows,] # remove those blankRows from dataFrame
+```
+
+Alternatively, you can use apply() to iterate through all rows. You can use this for a single data frame or a list of multiple data frames using a for loop. 
+```{r eval=F}
+# outputs indices of rows with all NA's
+blankRows <- which(apply(dataFrame,1,function(x)all(is.na(x)))) 
+```
+
+##Columns
+```{r eval=F}
+
+i=0
+allCols <- c()
+for(i in 1:length(dataFrame)){ # length(dataFrame) gives us # of cols
+  allRows[i] <- all(is.na(dataFrame[,i])) # notice that we switch where the i goes 
+}
+
+blankCols <- which(allCols)
+dataFrame <- dataFrame[,-blankCols]
+```
+
+Alternatively:
+```{r eval=F}
+blankCols <- which(apply(dataFrame,2,function(x)all(is.na(x))))
+```
+
+
+#Searching Through Strings - Dates
+
+Use the grepl() function to search for a particular string. Since we often have to reformat dates in our data sets, searching for particular dates or times could be useful. 
```

Comment:
Perhaps this would be a good place to introduce some helpful resources.  I personally like this cheatsheet: https://www.rstudio.com/wp-content/uploads/2016/09/RegExCheatsheet.pdf